linter/lints/common.go

```diff
@@ -40,46 +79,35 @@ func GetExtWithOID(exts []pkix.Extension, oid asn1.ObjectIdentifier) *pkix.Exten
 
 // ReadOptionalASN1BooleanWithTag attempts to read and advance incoming to
 // search for an optional DER-encoded ASN.1 element tagged with the given tag.
-// Unless outExpectedTagPresent or outPresent are nil, it stores whether the tag
-// was found in outExpectedTagPresent and an element with the tag was found in
-// outPresent, otherwise each boolean will take the default value. It reports
-// whether all reads were successful.
-func ReadOptionalASN1BooleanWithTag(incoming *cryptobyte.String, outExpectedTagPresent *bool, outPresent *bool, tag cryptobyte_asn1.Tag, defaultValue bool) bool {
-	if outExpectedTagPresent != nil {
-		*outExpectedTagPresent = defaultValue
-	}
-	if outPresent != nil {
-		*outPresent = defaultValue
-	}
-	if incoming.Empty() {
-		return false
-	}
-
-	// We need to check the boolean value to determine if the tag was found. The
-	// ReadOptionalASN1BooleanWithTag caller may not care about the value, but
-	// here internally we do.
-	tagPresent := false
-	var tagBytes cryptobyte.String
-
+// Unless out is nil, it stores whether an element with the tag was found in
+// out, otherwise out will take the default value. It reports whether all reads
+// were successful.
+func ReadOptionalASN1BooleanWithTag(incoming *cryptobyte.String, out *bool, tag cryptobyte_asn1.Tag, defaultValue bool) bool {
 	// ReadOptionalASN1 performs a peek and will not advance if the tag is
 	// missing, meaning that incoming will retain bytes.
-	ok := incoming.ReadOptionalASN1(&tagBytes, &tagPresent, tag)
-	if !ok && !tagPresent {
+	var valuePresent bool
+	var valueBytes cryptobyte.String
+	if !incoming.ReadOptionalASN1(&valueBytes, &valuePresent, tag) {
 		return false
 	}
-	if outExpectedTagPresent != nil && tagPresent {
-		*outExpectedTagPresent = true
-	}
-
-	// X.690 (07/2002) section 8.2 states that a boolean will have length of 1
-	// and value true will have contents FF.
-	// https://www.itu.int/rec/T-REC-X.690-200207-S/en
-	boolBytes := []byte(tagBytes)
-	if len(boolBytes) == 1 && boolBytes[0] == 0xFF {
-		if outPresent != nil {
-			*outPresent = true
+	val := defaultValue
+	if valuePresent {
+		boolBytes := []byte(valueBytes)
+		// X.690 (07/2002) section 8.2 states that a boolean will have length of 1
+		// and value true will have contents FF.
+		// https://www.itu.int/rec/T-REC-X.690-200207-S/en
```

Comment:
Can you update this comment with the more specific citations I gave in https://github.com/letsencrypt/boulder/pull/7064#discussion_r1340693208 ? Section 8.2.2 actually says that any nonzero value is true (and happens to give an _example_ where it's true is encoded as 0xFF); it's section 11.1 that says "in DER, true must be exactly 0xFF".